crates/hyperswitch_connectors/src/connectors/redsys/transformers.rs

```diff
@@ -1,228 +1,1499 @@
+    bill_addr_state: Option<Secret<String>>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ShippingData {
+    ship_addr_city: Option<String>,
+    ship_addr_country: Option<String>,
+    ship_addr_line1: Option<Secret<String>>,
+    ship_addr_line2: Option<Secret<String>>,
+    ship_addr_line3: Option<Secret<String>>,
+    ship_addr_postal_code: Option<Secret<String>>,
+    ship_addr_state: Option<Secret<String>>,
+}
+
+impl EmvThreedsData {
+    pub fn new(three_d_s_info: RedsysThreeDsInfo) -> Self {
+        Self {
+            three_d_s_info,
+            protocol_version: None,
+            browser_accept_header: None,
+            browser_user_agent: None,
+            browser_java_enabled: None,
+            browser_java_script_enabled: None,
+            browser_language: None,
+            browser_color_depth: None,
+            browser_screen_height: None,
+            browser_screen_width: None,
+            browser_t_z: None,
+            three_d_s_server_trans_i_d: None,
+            notification_u_r_l: None,
+            three_d_s_comp_ind: None,
+            cres: None,
+            billing_data: None,
+            shipping_data: None,
+        }
+    }
+
+    pub fn add_browser_data(mut self, browser_info: BrowserInformation) -> Result<Self, Error> {
+        self.browser_accept_header = Some(browser_info.get_accept_header()?);
+        self.browser_user_agent = Some(browser_info.get_user_agent()?);
+        self.browser_java_enabled = Some(browser_info.get_java_enabled()?);
+        self.browser_java_script_enabled = browser_info.get_java_script_enabled().ok();
+        self.browser_language = Some(browser_info.get_language()?);
+        self.browser_color_depth = Some(browser_info.get_color_depth()?.to_string());
+        self.browser_screen_height = Some(browser_info.get_screen_height()?.to_string());
+        self.browser_screen_width = Some(browser_info.get_screen_width()?.to_string());
+        self.browser_t_z = Some(browser_info.get_time_zone()?.to_string());
+        Ok(self)
+    }
+
+    pub fn set_three_d_s_server_trans_i_d(mut self, three_d_s_server_trans_i_d: String) -> Self {
+        self.three_d_s_server_trans_i_d = Some(three_d_s_server_trans_i_d);
+        self
+    }
+
+    pub fn set_protocol_version(mut self, protocol_version: String) -> Self {
+        self.protocol_version = Some(protocol_version);
+        self
+    }
+
+    pub fn set_notification_u_r_l(mut self, notification_u_r_l: String) -> Self {
+        self.notification_u_r_l = Some(notification_u_r_l);
+        self
+    }
+
+    pub fn set_three_d_s_comp_ind(mut self, three_d_s_comp_ind: ThreeDSCompInd) -> Self {
+        self.three_d_s_comp_ind = Some(three_d_s_comp_ind);
+        self
+    }
+
+    pub fn set_three_d_s_cres(mut self, cres: String) -> Self {
+        self.cres = Some(cres);
+        self
+    }
+
+    fn get_state_code(state: &str) -> Result<Secret<String>, Error> {
```

Comment:
This implementation has the same concerns pointed out in https://github.com/juspay/hyperswitch/pull/7273#discussion_r1957127718.